### PR TITLE
Migrate Example Snippets to Submodules

### PR DESCRIPTION
**Pull request by riastrad**
In order to ensure that example code is kept up to date with the latest working versions without having to do any copy/pasting, this PR adds submodules that can be updated to the latest versions via git. 

Remaining TODO:

- [x] Add submodules for Android's example repos
- [x] Point all examples pages to the proper submodule files
- [x] Cleanup any unnecessary copy/pasted code snippets in the repo
- [x] Update CONTRIBUTING.md with instructions on how to update submodules

### CODE REVIEW

**Comment by riastrad** [comment]
cc: @mapbox/android @mapbox/docs

**Comment by colleenmcginnis** [comment]
👋 @josh I caught this up to `mb-pages`. You should now be able to run locally after running `npm ci` (or removing `node_modules` and reinstalling). Let me know if you run into any trouble!

**Comment by riastrad** [comment]
beat me to it! thank you @colleenmcginnis! 💯

File: src/example-code/AddRainFallStyleActivity.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { highlightCodeSnippet } from '../util/highlight-code-snippet';
 
 function MakeCodeSnippetJava() {
-  const rawJavaCode = require('raw-loader!./java/add-rain-fall-style-activity--full-java.java');
+  const rawJavaCode = require('raw-loader!../../_includes/examples/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/AddRainFallStyleActivity.java');
```
Comment (colleenmcginnis) [inline src/example-code/AddRainFallStyleActivity.js]: Not the biggest deal, but the `_includes` folder for iOS was leftover from the Jekyll days. I'd recommend actually just putting these in a folder called `examples` in the base directory and then include `maps` and `navigation` folders in there. 🙂

**Comment by riastrad** [comment]
@colleenmcginnis Submodules should be all set once they've passed your official review! 🎉

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -142,9 +142,17 @@ All overview guides for APIs in the Mapbox Java SDK should follow this format:
...
 
-### Examples 
+### Examples
 
-The examples section contains **code examples**. Each example requires three pieces: (1) a code snippet, (2) visual assets, and (3) a Markdown file. 
+The examples section contains **code examples**. Examples are updated by bumping the either the `examples/maps` or `examples/navigation` submodule, based on the code in the [mapbox/mapbox-android-demo](https://github.com/mapbox/mapbox-android-demo) or the [mapbox/mapbox-navigation-android](https://github.com/mapbox/mapbox-navigation-android) repo. These repos contain working Android projects with apps that integrate the Java or Kotlin examples.
+
+#### Update examples
+1. Make and commit your changes in the proper repo.
+2. `git submodule update --remote examples/maps` (or `examples/navigation`) to update the submodule. This should typically point to the latest commit in master.
```
Comment (colleenmcginnis) [inline CONTRIBUTING.md]: Looks like the first time you pull in the the submodule you need to use `--init`. Does that sound right? I wonder if we should add that to the install instructions for the repo &mdash; I think we can use the recursive approach we use for our iOS docs.

Comment (riastrad) [reply CONTRIBUTING.md]: Ah, good catch. I've tested locally and it looks like cloning with the `--recursive` flag does the trick. I'll update the installation section of the `README.md` to explain this. 👍

**Comment by riastrad** [comment]
@colleenmcginnis changes are officially incorporated and tests are passing. Should be good for a quick final review.

**Comment by colleenmcginnis** [comment]
Thanks @riastrad! This looks good to me. Let's wait for a final  ✅ from @langsmith before merging -- just to make sure he has the chance to ask any questions about this new system before it's in production.

**Pull request merged**